src/components/QrcodeCapture.vue

```diff
@@ -13,18 +13,56 @@
       @change="onFileSelect"
     />
 
-    <img ref="imageElement" :src="imageBase64" :class="`${className}__image`" />
+    <canvas ref="canvasElement" :class="`${className}__image`" />
   </div>
 </template>
 
 <script>
 import { vibrate } from '@/lib/vibrate'
+import { useTemplateRef } from 'vue'
 
 export default {
   emits: ['detect', 'error'],
+  setup() {
+    const canvas = useTemplateRef('canvasElement')
+
+    const IMG_MAX_SIZE = 400
+
+    const drawCanvas = (file) => {
+      return new Promise((resolve, reject) => {
+        const img = new Image()
+
+        img.onload = () => {
+          const ctx = canvas.value.getContext('2d')
+
+          const ratio = img.width / img.height
+          let newWidth = IMG_MAX_SIZE
+          let newHeight = IMG_MAX_SIZE / ratio
+
+          if (newHeight > IMG_MAX_SIZE) {
+            newHeight = IMG_MAX_SIZE
+            newWidth = IMG_MAX_SIZE * ratio
+          }
+
+          canvas.value.width = newWidth
+          canvas.value.height = newHeight
+
+          ctx.drawImage(img, 0, 0, canvas.value.width, canvas.value.height)
+
+          resolve(canvas)
+        }
+
+        img.onerror = (e) => reject(e)
+
+        img.src = URL.createObjectURL(file)
```

Comment:
At some point you should call [URL.revokeObjectURL()](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static), otherwise the object will still exist even after user logsout.

I think you can revoke it right after the promise was resolved.